core/src/main/java/org/bitcoinj/core/MasternodeSync.java

```diff
@@ -108,13 +108,22 @@ public MasternodeSync(Context context)
             this.syncFlags = EnumSet.of(SYNC_DMN_LIST);
         else {
             this.syncFlags = EnumSet.noneOf(SYNC_FLAGS.class);
-            syncFlags.add(SYNC_MASTERNODE_LIST);
-            syncFlags.add(SYNC_GOVERNANCE);
-            syncFlags.add(SYNC_DMN_LIST);
+            //TODO:add other flags here to get other information such as governance messsages, by default
```

Comment:
#28 This should eliminate memory leaks with respect to MasternodeManager, which will not be used if one of the above flags is not set.